packages/net/net_address.pony

```diff
@@ -108,18 +108,76 @@ class val NetAddress is Equatable[NetAddress]
       recover String.from_cstring(consume serv) end)
 
   fun eq(that: NetAddress box): Bool =>
-    (this.length == that.length)
-      and (this.family == that.family)
-      and (this.port == that.port)
+      (this._family == that._family)
+      and (this._port == that._port)
       and (host_eq(that))
-      and (this.scope == that.scope)
+      and (this._scope == that._scope)
 
   fun host_eq(that: NetAddress box): Bool =>
     if ip4() then
-      this.addr == that.addr
+      this._addr == that._addr
+    else
+      (this._addr1 == that._addr1)
+        and (this._addr2 == that._addr2)
+        and (this._addr3 == that._addr3)
+        and (this._addr4 == that._addr4)
+    end
+
+  fun length() : U8 =>
+    """
+    For platforms (OSX/FreeBSD) with `length` field as part of 
+    its `struct sockaddr` definition, returns the `length`. 
+    Else (Linux/Windows) returns the size of `sockaddr_in` or `sockaddr_in6`.
+    """
+
+    ifdef linux or windows then
+      (@ponyint_address_length[U32](this)).u8()
     else
-      (this.addr1 == that.addr1)
-        and (this.addr2 == that.addr2)
-        and (this.addr3 == that.addr3)
-        and (this.addr4 == that.addr4)
+      ifdef bigendian then
+        ((_family >> 8) and 0xff).u8()
+      else
+        (_family and 0xff).u8()
+      end
     end
+
+    fun family() : U8 =>
+    """
+      Returns the `family`.
+    """
+
+    ifdef linux or windows then
+      ifdef bigendian then
+        ((_family >> 8) and 0xff).u8()
+      else
+        (_family and 0xff).u8()
+      end
+    else
+      ifdef bigendian then
+        (_family and 0xff).u8()
+      else
+        ((_family >> 8) and 0xff).u8()
+      end
+    end
+
+    fun port() : U16 =>
+      _port
+
+    fun scope() : U32 =>
+      _scope
+
+    fun addr() : (U32 | Array[U32]) =>
```

Comment:
what do you think about creating two different methods, on for the IPv4 and one for the IPv6 address? 
The user has to check what kind of address this is anyways, with a method able to return each, a second match statement necessary, like so:

```pony
let addr: NetAddress = ... // somehow get a  NetAddress
if addr.ip4() then
  match addr.addr()
  |  let ip4: U32 => // do something with ip4 address
  else
     // somehow handle this actually impossible case, otherwise this would always return something or None
  end
```

The `as` statement is also an alternative here. But after all, it might be more straightforward to do:

```pony
let addr: NetAddress = ... // somehow get a  NetAddress
if addr.ip4() then
  let ip4: U32 = addr.ip4_addr()
  // do something with ip4
end
```

It would need to be documented that the return value of a `fun ip4_addr(): U32` method is actually invalid if `ip4()` is not true. This would basically repeat the behavior we currently have with only the fields.